src/components/slider/slider.component.ts

```diff
@@ -330,8 +337,8 @@ export class SliderComponent implements OnInit, AfterViewInit, DoCheck {
         // get the element widths
         let thumbWidth: number;
 
-        if (this.options.handles.style === SliderStyle.Button) {
-            thumbWidth = this.options.track.height === SliderSize.Narrow ? 16 : 24;
+        if (this._options.handles.style === SliderStyle.Button) {
```

Comment:
I don't think this needed to change to `_options` because you defined an `options` getter.

Reply:
There is a performance benefit from accessing the property instead of the getter. We access options a lot, so by accessing the property directly we skip the intermediate function call:

https://stackblitz.com/edit/typescript-womh4x

Reply:
Actually, we also use `options` in the template, so ideally, `_options` would not be marked as `private` and the template should be updated to use `_options` instead. (https://www.reddit.com/r/Angular2/comments/6id6mf/low_performance_of_getters_binded_to_templates/)